WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift

```diff
@@ -1047,26 +1098,101 @@ private extension NotificationsViewController {
         }
 
-        guard let approveEnabled = block.action(id: ApproveCommentAction.actionIdentifier())?.enabled, approveEnabled == true else {
-            return rightButtons
+        if let approveAction = block.action(id: ApproveCommentAction.actionIdentifier()),
+            approveAction.enabled == true,
+            let button = approveAction.command?.icon as? MGSwipeButton {
+            button.callback = { _ in
+                let actionContext = ActionContext(block: block)
+                approveAction.execute(context: actionContext)
+                return true
+            }
+            rightButtons.append(button)
+        }
+
+        if FeatureFlag.notificationActions.enabled {
+            rightButtons.append(moreAction)
         }
 
-        let approveAction = block.action(id: ApproveCommentAction.actionIdentifier())
-        let button = approveAction?.command?.icon as? MGSwipeButton
+        return rightButtons
+    }
 
-        button?.callback = { _ in
-            let actionContext = ActionContext(block: block)
-            approveAction?.execute(context: actionContext)
-            return true
+    func showMoreAlertController(for note: Notification) {
+        let controller = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+
+        controller.addActionWithTitle(NSLocalizedString("Remind me...", comment: "Button title. Allows the user to pick a time to schedule a reminder."),
+                                      style: .default,
+                                      handler: { [weak self] action in
+            self?.showReminderAlert(for: note)
+        })
+
+        controller.addActionWithTitle(NSLocalizedString("Share...", comment: "Button title. Allows the user to share the content of a notification."),
+                                      style: .default,
+                                      handler: { [weak self] action in
+                                        self?.showActivityController(for: note)
+        })
+
+        controller.addCancelActionWithTitle(NSLocalizedString("Cancel", comment: "Button title. Dismisses an action sheet."))
+
+        configureViewControllerPopover(controller, forNote: note)
+        present(controller, animated: true, completion: nil)
+    }
+
+    func showActivityController(for note: Notification) {
+        guard let url = note.url else {
+            return
         }
 
-        rightButtons.append(button!)
+        var activityItems = [Any]()
+        if let subject = note.renderSubject() {
+            activityItems.append(subject as Any)
+        }
+        activityItems.append(url)
 
-        return rightButtons
+        let controller = UIActivityViewController(activityItems: activityItems, applicationActivities: nil)
+        configureViewControllerPopover(controller, forNote: note)
+        present(controller, animated: true, completion: nil)
     }
 }
 
 
+// MARK: - Reminders
+//
+private extension NotificationsViewController {
```

Comment:
I admire your courage - `NotificationsViewController` is not for the faint of heart. 😄